src/AutorestSwift/Code Generator/AutoRestHandler.swift

```diff
@@ -3,35 +3,51 @@ import NIO
 import os.log
 
 class AutoRestHandler {
-    private let logger: Logger
+    // MARK: Properties
 
-    init() {
-        self.logger = Logger(withFileName: "autorest-swift-debug.log")
+    private let plugin: AutoRestPlugin
+    private var logger: Logger {
+        return plugin.logger
     }
 
-    func handle(method: String, params _: RPCObject, callback: (RPCResult) -> Void) {
-        switch method.lowercased() {
-        case "getpluginnames":
-            logger.logToURL("Get a getpluginnames request")
-            return callback(.success(.list([.string("swift")]), RPCObjectType.response, ""))
-
-        case "process":
-            logger.logToURL("Get a process request")
+    // MARK: Initializers
 
-            let sessionId = "session_7"
+    public init(withPlugin plugin: AutoRestPlugin) {
+        self.plugin = plugin
+    }
 
-            logger.logToURL("Send back a process response")
-            callback(.success(.bool(true), RPCObjectType.response, ""))
+    // MARK: Methods
 
-            logger.logToURL("Send a ListInputs request")
-            return callback(.success(
-                .list([.string(sessionId), .string("code-model-v4")]),
-                RPCObjectType.request,
-                "ListInputs"
-            ))
+    func handle(method: String, params: RPCObject, callback: (RPCResult) -> Void) {
+        switch method.lowercased() {
+        case "getpluginnames":
+            callback(
+                .success(.list([.string("swift")]), .response, "")
+            )
+        case "process":
+            plugin.sessionId = params.asList?.last?.asString ?? ""
```

Comment:
I added the various `asFoo` calculated parameters to `RPCObject` to facilitate this.